lib/wraplibsnarkgadgets.cpp

```diff
@@ -49,18 +71,19 @@ std::string r1cs_to_json(protoboard<FieldT> pb, uint input_variables)
 
 void _sha256Constraints()
 {
-    // block_variable<FieldT> input;
-    // block_variable<FieldT> output;
-    // auto hash = std::make_shared<sha256_ethereum>(pb, 256, input; output, "cm_hash");
-
-    // std::shared_ptr<sha256_ethereum> hash;
     protoboard<FieldT> pb;
-    
-    // hash.reset(new sha256_ethereum(
-    //     pb, 256, input, output, "cm_hash"
-    // ));
-    // hash->generate_r1cs_constraints(true);
+    block_variable<FieldT> input(pb, 256, "input");
+    digest_variable<FieldT> output(pb, 256, "output");
+    //auto hash = std::make_shared<sha256_ethereum>(pb, 256, input, output, "hash");
```

Comment:
`std::make_shared` is neater and slightly faster (see [here](https://shaharmike.com/cpp/shared-ptr/#std-make-shared)) than the two stage construction you use below, would prefer to remove the lines below and uncomment this.